Tools/autotest/arduplane.py

```diff
@@ -6722,6 +6722,187 @@ def Volz(self):
             if abs(m.Pos - m.PosCmd) > 20:
                 break
 
+    def MountAutoDeploy(self):
+        '''Test Mount Auto-deploy'''
+
+        RETRACT_PWM = 1000
+        DEPLOY_PWM = 2000
+
+        self.set_parameter('MNT1_TYPE', 1) # Servo mount
+        self.set_parameter('MNT1_RETRACT_ALT', 200)
+        self.set_parameter('MNT1_DEPLOY_ALT', 250)
+        self.set_parameter('SERVO10_FUNCTION', 9) # k_mount_open
+        self.set_parameter('SERVO10_MIN', RETRACT_PWM)
+        self.set_parameter('SERVO10_MAX', DEPLOY_PWM)
+
+        self.reboot_sitl()
+
+        # grab home position:
+        self.mav.recv_match(type='HOME_POSITION', blocking=True)
+        home = self.mav.location()
+
+        # === Test auto deploy/retract in oscillating flight pattern ===
+        # If we're using auto deploy/retract, we expect to start in RETRACTED mode
+        self.assert_servo_channel_value(10, RETRACT_PWM)
+
+        self.context_collect("STATUSTEXT")
+        # Takeoff uses relative alt
+        self.takeoff(alt=300)
+
+        self.wait_statustext("Mount: Auto-deploy to default mode", check_context=True)
+        self.context_clear_collection("STATUSTEXT")
+        self.assert_servo_channel_value(10, DEPLOY_PWM)
+
+        # Check for retract message #1
+        self.context_collect("STATUSTEXT")
```

Comment:
Already doing this; clearing the collection doesn't stop collecting it ('though there is a `stop_collecting`)